### ISSUE TITLE
ALT+F menu in Google Chrome not announced, and NVDA+T reports "NVDA" instead of Chrome during initial load

### ISSUE THREAD

**Issue by amirsol81** [nvaccess/nvda#18694]
On two different machines, when pressing ALT+F in Google Chrome to open its main menu, NVDA does not announce the menu or any of its items. The menu does appear visually, and pressing the Up or Down arrow keys does move the focus between items, but NVDA remains silent.
Additionally, if Chrome is opened and NVDA+T is pressed while the first tab is still loading, sometimes NVDA, quite briefly, announces the word “NVDA” as if the focus were on the NVDA window instead of Chrome.
JAWS 2025 doesn't exhibit these issues.
### Steps to reproduce:
Open Google Chrome.
Press ALT+F to open the browser's main menu.
Also press Up/ Down, and observe whether NVDA announces anything.
### Actual behavior:
The Chrome menu opens, but NVDA remains silent, as if no menu appeared. Also pressing Up/ Down reads nothing.
### Expected behavior:
NVDA should announce that the Chrome menu has appeared and read its contents. Menu items should also be read when using Up/ Down.
#### NVDA installed/portable/running from source:
Both installed and portable, tested with NVDA 2024.4.2, portable, and various official and alpha releases made available over the past 4 months or so, up to NVDA Alpha 51856, installed.
#### NVDA version:
Various versions from 2024.4.2 official up to Alpha 51856/
#### Windows version:
Windows 11 24H2
#### Name and version of other software in use when reproducing the issue:
Google Chrome V139. but duplicable with various older Chrome releases at least up to V133
#### Does the issue still occur after restarting your computer?
Yes.
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes - mentioned above
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by SaschaCowley** [comment]
@amirsol81 I'm able to reproduce this issue. As a workaround, it seems like switching to focus mode (`NVDA+space`) allows the arrow keys to move in the menu.

**Comment by amirsol81** [comment]
> [@amirsol81](https://github.com/amirsol81) I'm able to reproduce this issue. As a workaround, it seems like switching to focus mode (`NVDA+space`) allows the arrow keys to move in the menu.

@SaschaCowley Thanks for sharing the work-around. Have you also noticed that as we press Control+T to type a new URL, NVDA strangely says "Address and search bar  edit  Search Google or type a URL  A?" I mean what is that "A"? Pressing Left/ Right also displays that "A" in the edit box - an "A" which doesn't really exist. So NVDA can't tell us that the edit box is blank.

**Comment by seanbudd** [comment]
Is this a recent regression in NVDA? Could you reproduce this with 2025.1 or an older version?

**Comment by amirsol81** [comment]
> Is this a recent regression in NVDA? Could you reproduce this with 2025.1 or an older version?

@seanbudd As mentioned in the bug report, the issues can be duplicated with NVDA releases as old as 2024.4.2. I haven't tested releases beyond that.

**Comment by seanbudd** [comment]
Can you please split these into separate issues? Please open a new issue for the NVDA+T issue

**Comment by CyrilleB79** [comment]
@seanbudd the issue occurs on NVDA 2025.3beta2 (installed) and 2025.1 (launcher) and was not present more 3 weeks ago. I am sure that it was not present on NVDA 2025.1 when I had it installed on my computer. Thus I suspect a Chrome update.

**Comment by seanbudd** [comment]
Since this might be a regression in Chrome, has anyone reported this to the Chrome developers? Can this also be reproduced in edge?

**Comment by SaschaCowley** [comment]
I can't repro it in Edge, but Edge also handles focus differently to Chrome in my experience

**Comment by CyrilleB79** [comment]
I have opened this [bug report](https://issues.chromium.org/issues/441317307).
Let's hope that's correct; that's the first time I've opened a bug report for Chrome.

**Comment by amirsol81** [comment]
@SaschaCowley @seanbudd I wanted to separate the issues, but something else, which is potentially related, should be mentioned. NVDA has issues reliably opening context menus on links in Google Chrome when the Applications key is pressed. Sometimes the menu appears briefly and disappears, requiring Escape before trying again. This began a couple of months ago, just as we began encountering the ALT+F issue mentioned here.

**Comment by Adriani90** [comment]
Testing with Chrome 144.0.7523.0 (Offizieller Build) canary (64-Bit)
And NVDA alpha-53273,a6a536eb (2026.1.0.53273)

1. Issue with the menu (alt+f or alt+e) does not occur on my end. Pressing up and down arrow works properly. This shortcut works only when in focus mode.
2. Issue with nvda+t reports properly the title of the website opened along with "Google Chrome" which is expected behavior
3. Context menu appears reliably when pressing context menu most of the time. However, sometimes I experience the context menu disappearing as described in this issue. In this attached log, it seems there are some double focus events:

```
DEBUGWARNING - UIAHandler.UIAHandler.IUIAutomationFocusChangedEventHandler_HandleFocusChangedEvent (00:52:52.972) - Dummy-17 (6228):
HandleFocusChangedEvent: Ignoring duplicate focus event 
```

This seems to happen always when the context menu disappears or when NVDA is silent after pressing down arrow for the first time in the context menu.

[nvda_chrome.txt](https://github.com/user-attachments/files/23488112/nvda_chrome.txt)

**Comment by amirsol81** [comment]
> 1. Issue with the menu (alt+f or alt+e) does not occur on my end. Pressing up and down arrow works properly. This shortcut works only when in focus mode.

@Adriani90 That's the point. It should definitely work in Browse mode, as it did in the past, and as it reliably does with JAWS.

**Comment by benbeaudry** [comment]
Hi all! I just fixed this issue in Chrome: https://chromium-review.googlesource.com/c/chromium/src/+/7147587. The fix should be available for testing in tomorrow's **Chrome Canary**, and roll out to all users over the coming weeks. Thanks for reporting the issue.

**Comment by seanbudd** [comment]
Thanks for the fix @benbeaudry !

**Comment by amirsol81** [comment]
@seanbudd With all respect, why have you closed this before even seeing the fix in action? Yes it's great to know it's going to be fixed, but it hasn't reached us yet. It's safe practice to see the fix first before closing issues like this. One fewer issue feels good - that I know. But this is not logical IMHO.

**Comment by seanbudd** [comment]
@amirsol81 - we close issues when fixes are merged and released to alpha software for testing. We can always reopen this if further issues are found.
We close NVDA issues under the same logic, when a PR is merged to master and released on alpha, not when it has had testing on alpha or stable.

**Comment by amirsol81** [comment]
@seanbudd As far as what @benbeaudry said is concerned, this will  reach Chrome Canary today - as we speak. So it's supposed to reach alpha-level testing today. I think it's not appropriate to close issues like this, but it's just my two cents.

**Comment by benbeaudry** [comment]
Fix is now publicly available in Chrome 144.0.7527.0 (Canary). Verified as fixed.

**Comment by Adriani90** [comment]
I confirm this works now as expected, thank you very much for the quick fix.

 

 

Von: Benjamin Beaudry ***@***.***> 
Gesendet: Freitag, 14. November 2025 18:04
An: nvaccess/nvda ***@***.***>
Cc: Adriani90 ***@***.***>; Mention ***@***.***>
Betreff: Re: [nvaccess/nvda] ALT+F menu in Google Chrome not announced, and NVDA+T reports "NVDA" instead of Chrome during initial load (Issue #18694)

 

  <https://avatars.githubusercontent.com/u/59941307?s=20&v=4> benbeaudry left a comment (nvaccess/nvda#18694) <https://github.com/nvaccess/nvda/issues/18694#issuecomment-3533721934> 

Fix is now publicly available in Chrome 144.0.7527.0 (Canary). Verified as fixed.
